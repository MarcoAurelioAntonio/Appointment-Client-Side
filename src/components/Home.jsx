import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getAllCoursesApi } from '../redux/coursesSlice';
import Courses from './Courses';
import NavMenu from './NavMenu';

const Home = () => {
  const { courses, loading } = useSelector((store) => store.courses);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getAllCoursesApi());
  }, []);

  return (
    <section className="home-container">
      <NavMenu bgColor="green" isBacking={false} isHide={false} />
      {
        loading && <p>Loading...</p>
      }
      <div className="home-wrapper">
        <div className="over-wrapper">
          <h1 className="home-title">LATEST COURSES</h1>
          <h5>Please select a course</h5>
          <div className="separator">. . . . . . . . . . .</div>
        </div>
        {
          courses.length === 0 ? (
            <p>No courses available</p>
          ) : (
            <Courses courses={courses} />
          )
        }
      </div>
    </section>
  );
};

export default Home;
