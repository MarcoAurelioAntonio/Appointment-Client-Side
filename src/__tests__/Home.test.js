import { render, screen } from '@testing-library/react';
import { configureStore, createSlice } from '@reduxjs/toolkit';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Home from '../components/Home';

describe('Tests for <Home /> component', () => {

  const user = {
    id: 1,
    name: 'Nelson',
  };

  const userSlice = createSlice({
    name: 'users',
    initialState: {
      current_user: user,
      status: 'idle',
      loading: false,
      error: null,
    }
  });

  const coursesList = [{
    "id": 9,
    "name": "Nelson asdadsasdasdads",
    "description": "dsadasdasdasdasdasdadsasdasdasdasddas",
    "start_date": "2023-06-21",
    "end_date": "2023-06-24",
    "price": "20.0",
    "course_type": "asdads",
    "image_url": "http://localhost:3000/rails/active_storage/blobs/redirect/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBDZz09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--b4fee963a3f0b6b5be716592d385e19264fa88ca/Screenshot%202023-06-20%20at%2012.44.39%20PM.png"
  },
  {
    "id": 10,
    "name": "THis is new from zoom",
    "description": "This is a test from the last meeting",
    "start_date": "2023-06-22",
    "end_date": "2023-06-28",
    "price": "45.0",
    "course_type": "Biology",
    "image_url": "http://localhost:3000/rails/active_storage/blobs/redirect/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBDdz09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--e03ef93d475cce1f32c6ec7858b45fb6bcd17fb5/Simulator%20Screenshot%20-%20iPhone%2014%20Pro%20Max%20-%202023-06-16%20at%2018.28.01.png"
  },
  {
    "id": 11,
    "name": "Last test",
    "description": "This is a description asdadsasd",
    "start_date": "2023-06-22",
    "end_date": "2023-06-30",
    "price": "45.0",
    "course_type": "Computing",
    "image_url": "http://localhost:3000/rails/active_storage/blobs/redirect/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBEQT09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--232f32bad1eac6bebb8856b40864e32215de0447/Screenshot%202023-06-20%20at%2012.44.39%20PM.png"
  }];

  const coursesSlice = createSlice({
    name: 'courses',
    initialState: {
      courses: coursesList,
      course: {},
      status: 'idle',
      loading: false,
      error: null,
      deleted: false,
    }
  });

  const store = configureStore({
    reducer: {
      users: userSlice.reducer,
      courses: coursesSlice.reducer,
    },
  });

  it('In the home screen should be displayed 3 courses', () => {
    render(
      <BrowserRouter>
        <Provider store={store}>
          <Home />
        </Provider>
      </BrowserRouter>,
    );
    const element = screen.getByTestId('courses-item');
    expect(element.children.length).toBe(3);
  })
  it('Should match the snapshot', () => {
    expect(renderer.create(<BrowserRouter>
      <Provider store={store}>
        <Home />
      </Provider>
    </BrowserRouter>,).toJSON()).toMatchSnapshot();
  });
});
