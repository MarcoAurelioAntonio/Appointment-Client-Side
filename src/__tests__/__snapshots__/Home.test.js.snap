// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <Home /> component Should match the snapshot 1`] = `
<section
  className="home-container"
>
  <div
    className="nav-container"
  >
    <div
      className="imgcontainer"
    >
      <img
        alt="Logo"
        className="avatar"
        src="global.png"
      />
    </div>
    <div>
      <div
        className="nav_item active"
      >
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          COURSES
        </a>
      </div>
      <div
        className="nav_item notActive"
      >
        <a
          className=""
          href="/add-reservation"
          onClick={[Function]}
        >
          ENROLL
        </a>
      </div>
      <div
        className="nav_item notActive"
      >
        <a
          className=""
          href="/all_user_reservations"
          onClick={[Function]}
        >
          MY ENROLLMENTS
        </a>
      </div>
      <div
        className="nav_item notActive"
      >
        <a
          className=""
          href="/add_course"
          onClick={[Function]}
        >
          ADD COURSE
        </a>
      </div>
      <div
        className="nav_item notActive"
      >
        <a
          className=""
          href="/delete_course"
          onClick={[Function]}
        >
          DELETE COURSE
        </a>
      </div>
    </div>
    <button
      className="close-btn"
      onClick={[Function]}
      type="button"
    >
      <span
        className="material-symbols-outlined back-btn"
      >
        arrow_left
      </span>
    </button>
    <div
      className="footer"
    >
      <div
        className="social-icons"
      >
        <a
          href="https://github.com/MarcoAurelioAntonio/Global-Academy-Client"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-twitter "
            data-icon="twitter"
            data-prefix="fab"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <i
            className="px-2"
          />
        </a>
        <a
          href="https://github.com/MarcoAurelioAntonio/Global-Academy-Client"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-facebook-f "
            data-icon="facebook-f"
            data-prefix="fab"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <i
            className="px-2"
          />
        </a>
        <a
          href="https://github.com/MarcoAurelioAntonio/Global-Academy-Client"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-google "
            data-icon="google"
            data-prefix="fab"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 488 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M488 261.8C488 403.3 391.1 504 248 504 110.8 504 0 393.2 0 256S110.8 8 248 8c66.8 0 123 24.5 166.3 64.9l-67.5 64.9C258.5 52.6 94.3 116.6 94.3 256c0 86.5 69.1 156.6 153.7 156.6 98.2 0 135-70.4 140.8-106.9H248v-85.3h236.1c2.3 12.7 3.9 24.9 3.9 41.4z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <i
            className="px-2"
          />
        </a>
        <a
          href="https://github.com/MarcoAurelioAntonio/Global-Academy-Client"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-vimeo "
            data-icon="vimeo"
            data-prefix="fab"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M403.2 32H44.8C20.1 32 0 52.1 0 76.8v358.4C0 459.9 20.1 480 44.8 480h358.4c24.7 0 44.8-20.1 44.8-44.8V76.8c0-24.7-20.1-44.8-44.8-44.8zM377 180.8c-1.4 31.5-23.4 74.7-66 129.4-44 57.2-81.3 85.8-111.7 85.8-18.9 0-34.8-17.4-47.9-52.3-25.5-93.3-36.4-148-57.4-148-2.4 0-10.9 5.1-25.4 15.2l-15.2-19.6c37.3-32.8 72.9-69.2 95.2-71.2 25.2-2.4 40.7 14.8 46.5 51.7 20.7 131.2 29.9 151 67.6 91.6 13.5-21.4 20.8-37.7 21.8-48.9 3.5-33.2-25.9-30.9-45.8-22.4 15.9-52.1 46.3-77.4 91.2-76 33.3.9 49 22.5 47.1 64.7z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <i
            className="px-2"
          />
        </a>
        <a
          href="https://github.com/MarcoAurelioAntonio/Global-Academy-Client"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-pinterest "
            data-icon="pinterest"
            data-prefix="fab"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 496 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M496 256c0 137-111 248-248 248-25.6 0-50.2-3.9-73.4-11.1 10.1-16.5 25.2-43.5 30.8-65 3-11.6 15.4-59 15.4-59 8.1 15.4 31.7 28.5 56.8 28.5 74.8 0 128.7-68.8 128.7-154.3 0-81.9-66.9-143.2-152.9-143.2-107 0-163.9 71.8-163.9 150.1 0 36.4 19.4 81.7 50.3 96.1 4.7 2.2 7.2 1.2 8.3-3.3.8-3.4 5-20.3 6.9-28.1.6-2.5.3-4.7-1.7-7.1-10.1-12.5-18.3-35.3-18.3-56.6 0-54.7 41.4-107.6 112-107.6 60.9 0 103.6 41.5 103.6 100.9 0 67.1-33.9 113.6-78 113.6-24.3 0-42.6-20.1-36.7-44.8 7-29.5 20.5-61.3 20.5-82.6 0-19-10.2-34.9-31.4-34.9-24.9 0-44.9 25.7-44.9 60.2 0 22 7.4 36.8 7.4 36.8s-24.5 103.8-29 123.2c-5 21.4-3 51.6-.9 71.2C65.4 450.9 0 361.1 0 256 0 119 111 8 248 8s248 111 248 248z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <i
            className="px-2"
          />
        </a>
      </div>
      <div
        className="text"
      >
        <span>
          Â© 2023 Global Academy
        </span>
      </div>
    </div>
  </div>
  <div
    className="home-wrapper"
  >
    <div
      className="over-wrapper"
    >
      <h1
        className="home-title"
      >
        LATEST COURSES
      </h1>
      <h5>
        Please select a course
      </h5>
      <div
        className="separator"
      >
        . . . . . . . . . . .
      </div>
    </div>
    <div
      className="carousel-courses css-ohwg9z"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <div
        className="css-1f8sh1y"
        style={
          Object {
            "height": undefined,
          }
        }
      >
        <div
          className="css-1l7c0cy"
        >
          <div
            draggable={false}
            style={
              Object {
                "WebkitTouchCallout": "none",
                "WebkitUserSelect": "none",
                "height": "100%",
                "touchAction": "pan-y",
                "userSelect": "none",
              }
            }
          >
            <div
              style={
                Object {
                  "height": "100%",
                  "position": "relative",
                }
              }
            >
              <div
                style={
                  Object {
                    "height": undefined,
                  }
                }
              >
                <div
                  className="course-group"
                  data-testid="courses-item"
                >
                  <a
                    className="course-item"
                    href="/courses/9"
                    onClick={[Function]}
                  >
                    <div
                      className="image-container"
                    >
                      <img
                        alt="Course-img"
                        className="course-image-nelson"
                        src=""
                      />
                    </div>
                    <p
                      className="course-title"
                      data-testid="name-9"
                    >
                      Nelson asdadsasdasdads
                    </p>
                    <div
                      className="separator"
                    >
                      . . . . . . . . . . .
                    </div>
                    <p
                      className="course-label"
                    >
                      dsadasdasdasdasdasdadsasdasdasdasddas
                    </p>
                    <p
                      className="course-price"
                    >
                      Course-Type: asdads
                    </p>
                    <p
                      className="course-price"
                    >
                      Price: 20.0 $
                    </p>
                    <div
                      className="course-duration"
                    >
                      <p
                        className="course-price"
                      >
                        Duration: 0 hours
                      </p>
                    </div>
                  </a>
                  <a
                    className="course-item"
                    href="/courses/10"
                    onClick={[Function]}
                  >
                    <div
                      className="image-container"
                    >
                      <img
                        alt="Course-img"
                        className="course-image-nelson"
                        src=""
                      />
                    </div>
                    <p
                      className="course-title"
                      data-testid="name-10"
                    >
                      THis is new from zoom
                    </p>
                    <div
                      className="separator"
                    >
                      . . . . . . . . . . .
                    </div>
                    <p
                      className="course-label"
                    >
                      This is a test from the last meeting
                    </p>
                    <p
                      className="course-price"
                    >
                      Course-Type: Biology
                    </p>
                    <p
                      className="course-price"
                    >
                      Price: 45.0 $
                    </p>
                    <div
                      className="course-duration"
                    >
                      <p
                        className="course-price"
                      >
                        Duration: 0 hours
                      </p>
                    </div>
                  </a>
                  <a
                    className="course-item"
                    href="/courses/11"
                    onClick={[Function]}
                  >
                    <div
                      className="image-container"
                    >
                      <img
                        alt="Course-img"
                        className="course-image-nelson"
                        src=""
                      />
                    </div>
                    <p
                      className="course-title"
                      data-testid="name-11"
                    >
                      Last test
                    </p>
                    <div
                      className="separator"
                    >
                      . . . . . . . . . . .
                    </div>
                    <p
                      className="course-label"
                    >
                      This is a description asdadsasd
                    </p>
                    <p
                      className="course-price"
                    >
                      Course-Type: Computing
                    </p>
                    <p
                      className="course-price"
                    >
                      Price: 45.0 $
                    </p>
                    <div
                      className="course-duration"
                    >
                      <p
                        className="course-price"
                      >
                        Duration: 0 hours
                      </p>
                    </div>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className=" css-1m9128y"
        style={Object {}}
      >
        <button
          aria-label="carousel indicator 1"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium   css-8ucetn-MuiButtonBase-root-MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-aq8pb7-MuiSvgIcon-root"
            data-testid="FiberManualRecordIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="8"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</section>
`;
